### DISCUSSION TITLE
There should be empty() function

### DISCUSSION THREAD

**Discussion by subhasundardass** [rrousselGit/riverpod#1465] [Ideas]
It will be very helpful if there is an empty function at AsyncValue when.

R when<R>({
  required R Function(Model) data,
  required R Function(Object, StackTrace?) error,
  required R Function() loading,
  
empty ****
})

**Comment by rrousselGit** [comment]
What exactly is it supposed to do?

**Reply by subhasundardass** [reply]
if api respond with empty List.  
like.. List Cart = []

**Reply by rrousselGit** [reply]
Why not check that inside `data`?

```dart
ref.watch(provider).when(
  data: (value) {
    if (value.isEmpty) {...}
  }
)
```

**Reply by subhasundardass** [reply]
Yes, it can be, but I was looking something like empty function, Like
body: controller.obx(
        (state)=>Text(state.name),
        
        // here you can put your custom loading indicator, but
        // by default would be Center(child:CircularProgressIndicator())
        onLoading: CustomLoadingIndicator(),
        onEmpty: Text('No data found'),

        // here also you can set your own error widget, but by
        // default will be an Center(child:Text(error))
        onError: (error)=>Text(error),
      ),
